cherryontech/jupiter-jumpers
# filter functional, fixed a bug as well

## Description

 This component is integrated into the "trends" page to allow users to select a date range for updating charts dynamically based on the selected date range.


_Feature update/change/addition, bug fix, styling update, etc._

 

- [ ] The date range picker component should be functional and display correctly.

 

- [ ] The selected date range should be displayed in the button as per the user's selection.

 

- [ ] Toggling the button should show/hide the date range picker.

 

- [ ] Charts on the "trends" page should update dynamically based on the selected date range.



## Update Screenshot
<img width="365" alt="Screenshot 2024-06-06 at 10 20 14 AM" src="https://github.com/cherryontech/Uncloud/assets/123599985/07a44a9f-cdac-4131-a17a-cd72019cf15b">



## Testing Instructions

1. Open the "trends" page in the application.
2. Ensure that there is a select button to choose a custom date range or predefined buttons for week, month, and last month for comparison.
3. Click on the select button to display the date range picker.
4. Select a custom date range using the calendar interface or click on one of the predefined buttons for week, month, or last month.
5. Verify that the selected date range is displayed in the button.
6. Click on the button again to hide the date range picker.
7. Verify that the charts on the "trends" page update dynamically based on the selected date range.
8. Add moods, logs, wins, favorites, and frequented words to the application.
9. Keep an eye on the changes in all trend charts corresponding to the added data.


## Learnings (Optional)
react-date-range library
